libraries/resource_provider.rb

```diff
@@ -221,6 +221,16 @@ def get_diff(resource, desired_data)
     def recursive_diff(data, desired_data, str = '', indent = '')
       OneviewCookbook::Helper.recursive_diff(data, desired_data, str, indent)
     end
+
+    # Generic method to retrieve and return a resource from different resources using the type and name of the resource
+    # @param [String, Symbol] resource_class_type Type of resource to be retrieved. e.g., :GoldenImage, :FCNetwork.
+    # @param [String] resource_id Name for this resource. 'Resource1'
+    def load_resource_uri(resource_class_type, resource_id)
+      return unless resource_id
+      resource_instance = resource_named(resource_class_type).new(@item.client, name: resource_id)
+      raise "#{resource_class_type} resource with name '#{resource_id}' was not found in the appliance." unless resource_instance.retrieve!
+      resource_instance['uri']
```

Comment:
This sure makes it easy to use, and I like it for this resource, but can I suggest just 1 more thing? What if it took an additional parameter, which describes what it should return. ie
```ruby
def load_resource(resource_class_type, resource_id, ret_attribute = 'uri')
# and maybe even return the entire resource if ret_attribute is nil or a special value like :resource
```
That way it can be used to return things other than just the uri